main.tf

```diff
@@ -66,23 +62,12 @@ resource "aws_rds_cluster_instance" "aurora_db_delete" {
   }
 }
 
-resource "aws_rds_cluster_instance" "aurora_db_no_delete" {
-  count              = var.deletion_protection ? var.db_instance_count : 0
-  identifier         = "${var.db_name}-instance-${count.index + 1}"
-  cluster_identifier = aws_rds_cluster.aurora_cluster.cluster_identifier
-
-  publicly_accessible = false
-
-  engine_version = var.engine_version
-  engine         = var.engine
-
-  db_subnet_group_name = aws_db_subnet_group.rds_subnet_group.name
-  instance_class       = var.db_instance_class
-
-  performance_insights_enabled = var.performance_insights_enabled
+moved {
+  from = aws_rds_cluster_instance.aurora_db_delete
+  to   = aws_rds_cluster_instance.aurora_db
+}
```

Comment:
Will the `moved` block cause any issues if the consumer of this module doesn't already have a no-delete db instance in place?

Reply:
No, shouldn't be a problem. Since it will rename both resources to a new name, once they're renamed they would no longer be there anyway. Therefore, the moved block must be fine when the `from` is not found.